### PR TITLE
Add support for pmtiles and cog

### PR DESCRIPTION
**Pull request by eliotjordan**
- Replace webpacker with vite and add openlayers dependenciework on conversion to vites
- Add openlayers viewer for pmtiles and cog
- Refactor geocoder plugin and switch to ESRI provider
- Replace eslint with standard js

Work towards: https://github.com/pulibrary/figgy/issues/5850

### CODE REVIEW

File: config/initializers/monkeypatches/download.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+module Geoblacklight
+  class Download
```
Comment (tpendragon) [inline config/initializers/monkeypatches/download.rb:2]: Suggestion: My assumption reading this is that all these methods are overridden and customized completely, if that's not true can we have some comments sharing which are overridden and how?

File: config/initializers/monkeypatches/item_viewer.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+# Overridden from:
+# https://github.com/geoblacklight/geoblacklight/blob/main/lib/geoblacklight/item_viewer.rb
```
Comment (tpendragon) [inline config/initializers/monkeypatches/item_viewer.rb:4]: Question: Why'd we have to override this?

**Review by tpendragon — CHANGES_REQUESTED** [review]
Just a few questions. Otherwise, looks great! I admit I don't totally understand the openlayers JS.

File: config/initializers/monkeypatches/download.rb (hunk 1 above)
Comment (eliotjordan) [reply config/initializers/monkeypatches/download.rb:2]: This is an older monkey patch that I moved to this location. Yes we can add comments.

**Review by hackartisan — DISMISSED** [review]
Can this ref an issue? It would be helpful to record why we are doing this.

File: config/initializers/monkeypatches/download.rb (hunk 1 above)
Comment (eliotjordan) [reply config/initializers/monkeypatches/download.rb:2]: Actually, already has a comment. I'll add link to the file that was overridden.

Comment (eliotjordan) [reply config/initializers/monkeypatches/download.rb:3]: Link added.

File: config/initializers/monkeypatches/item_viewer.rb (hunk 2 above)
Comment (eliotjordan) [reply config/initializers/monkeypatches/item_viewer.rb:4]: Comments added.

File: config/initializers/monkeypatches/constants.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# Overridden from:
+# https://github.com/geoblacklight/geoblacklight/blob/main/lib/geoblacklight/constants.rb
```
[the remark this answers is not in the document]
Comment (eliotjordan) [reply config/initializers/monkeypatches/constants.rb:4]: Comment added.

**Comment by eliotjordan** [comment]
@hackartisan Reffed issue.

**Review by tpendragon — APPROVED** [review]
This looks good to me. Part of me wishes that the eslint -> standardjs and the vite stuff were different PRs, but c'est la vie.

**Comment by eliotjordan** [comment]
@tpendragon When project don't get a lot of time or attention, technical debt accumulates. Both the eslint and vite changes made sense at the time and were hard to untangle.

**Pull request merged**